pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -511,10 +511,10 @@ export default class SettingsValidator {
       return false;
     }
 
-    let changed = false;
+    let changed = Object.keys(currentValue).some(k => !(k in desiredValue));
```

Comment:
If you're setting `changed` here couldn't it be a constant, and we don't need the `changed ||= ...` bit at line 520 ?

Reply:
This only covers where the current value is a superset of desired value; the other part covers where they both have a key, but it's different.